### PR TITLE
[plotly.js] Update types for x and y axis names to match documentation

### PR DESCRIPTION
**Pull request by mrtnbrst**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

**I did not run the tests as the command failed even before running the test** 

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://plotly.com/python/reference/layout/shapes/#layout-shapes-items-shape-xref, https://plotly.com/python/reference/layout/shapes/#layout-shapes-items-shape-yref
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

I added myself to the contributors as I contributed changes to the plotly.js type definitions [three times ](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/master/?author=mrtnbrst) by now and I will continue to do so once I face an error.

### CODE REVIEW

**Comment by mrtnbrst** [comment]
As mentioned above: **I did not run the tests as the command failed even before running the test**

**Comment by jakebailey** [comment]
> I did not run the tests as the command failed even before running the test

> As mentioned above: I did not run the tests as the command failed even before running the test

I'm not sure you've posted enough info for someone to help with that; if you do, I can tell you where things are going wrong.

File: types/plotly.js/index.d.ts [hunk 1]
```diff
@@ -842,8 +842,16 @@ export type Calendar =
...
-export type YAxisName = "y" | "y2" | "y3" | "y4" | "y5" | "y6" | "y7" | "y8" | "y9" | "y10" | "y11";
+// regex from documentation: "/^x([2-9]|[1-9][0-9]+)?( domain)?$/" | "/^y([2-9]|[1-9][0-9]+)?( domain)?$/"
+// regex allows for an unlimited amount of digits for the 'axis number', but the following typescript definition is limited to two digits
+type xYAxisNames = `${
+    | ""
+    | `${2 | 3 | 4 | 5 | 6 | 7 | 8 | 9}`
+    | `${1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9}${0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9}`}${"" | " domain"}`;
+
+export type XAxisName = `${"x"}${xYAxisNames}`;
+export type YAxisName = `${"y"}${xYAxisNames}`;
```
Comment (jakebailey) [inline types/plotly.js/index.d.ts:853]:
```suggestion
export type XAxisName = `x${xYAxisNames}`;
export type YAxisName = `y${xYAxisNames}`;
```

Not sure this needs to have these extra ones.

**Comment by mrtnbrst** [comment]
> > I did not run the tests as the command failed even before running the test
> 
> > As mentioned above: I did not run the tests as the command failed even before running the test
> 
> I'm not sure you've posted enough info for someone to help with that; if you do, I can tell you where things are going wrong.

Sorry for my late reply - I'll post more details by the end of the week.

**Review by sandersn — APPROVED** [review]

**Comment by mrtnbrst** [comment]
`node ./scripts/clean-node-modules.js 
pnpm install --filter plotly.js
pnpm test plotly.js`

This leads to:

OS: Windows 10, node version 22.3

> pnpm test plotly.js
> 
> > definitely-typed@0.0.3 test C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped
> > node --enable-source-maps node_modules/@definitelytyped/dtslint/ types "plotly.js"
> 
> node:internal/modules/cjs/loader:1215
>   throw err;
>   ^
> 
> Error: Cannot find module 'C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped\node_modules\@definitelytyped\dtslint'
>     at Module._resolveFilename (node:internal/modules/cjs/loader:1212:15)
>     at Module._load (node:internal/modules/cjs/loader:1038:27)
>     at wrapModuleLoad (node:internal/modules/cjs/loader:212:19)
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:5)
>     at node:internal/main/run_main_module:30:49 {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: []
> }
> 
> Node.js v22.3.0
>  ELIFECYCLE  Test failed. See above for more details.
>  WARN   Local package.json exists, but node_modules missing, did you mean to install?
> 
> C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped>pnpm install --filter plotly.js
> Done in 14s
> 
> C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped>pnpm install --filter plotly.js
> Done in 15.9s
> 
> C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped>pnpm test plotly.js
> 
> > definitely-typed@0.0.3 test C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped
> > node --enable-source-maps node_modules/@definitelytyped/dtslint/ types "plotly.js" 
> 
> node:internal/modules/cjs/loader:1215
>   throw err;
>   ^
> 
> Error: Cannot find module 'C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped\node_modules\@definitelytyped\dtslint'
>     at Module._resolveFilename (node:internal/modules/cjs/loader:1212:15)
>     at Module._load (node:internal/modules/cjs/loader:1038:27)
>     at wrapModuleLoad (node:internal/modules/cjs/loader:212:19)
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:5)
>     at node:internal/main/run_main_module:30:49 {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: []
> }
> 
> Node.js v22.3.0
>  ELIFECYCLE  Test failed. See above for more details.
>  WARN   Local package.json exists, but node_modules missing, did you mean to install?


Ok, then I installed dslint manually using `npm install dslint`

`pnpm test plotly.js` does not throw an error then, but it apparently does not actually run the tests.
`
pnpm test plotly.js

> definitely-typed@0.0.3 test C:\Users\Admin\Documents\vscodeprojects\DefinitelyTyped
> node --enable-source-maps node_modules/@definitelytyped/dtslint/ types "plotly.js"

dtslint@0.2.22
`

 A `throw new Error()` in the beginning of a test (e.g. https://github.com/mrtnbrst/DefinitelyTyped/blob/ac03da650c909c9fe0bf044f92efc58cfe930772/types/plotly.js/test/index-tests.ts#L12)  does not show any effect 😞.

What step of the README did i get wrong?

Thanks for you support and have a nice weekend!

**Comment by sandersn** [comment]
dtslint is not running the test code, only compiling it. Compile errors will fails the tests.

Confusingly, `dtslint` is the old package. `@definitelytyped/dtslint` is the new one. `pnpm -w install` should install packages for the whole DT repo. (Or `pnpm install` at the DT root.)

**Comment by mrtnbrst** [comment]
I assumed it would run the code.

I am supposed to run it [README](https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#testing):

> Create types/foo/index.d.ts containing declarations for the module "foo". You should now be able to import from "foo" in your code and it will route to the new type definition. Then build and run the code to make sure your type definition actually corresponds to what happens at runtime.

How am I supposed to run the code then?
On second thought I think especially for plotly.js it does not properly work with node.js and requires a browser.

@sandersn as there are no compile errors, I assume that I can actually merge my changes?

**Comment by mrtnbrst** [comment]
@sandersn should I open an issue so someone can mark https://www.npmjs.com/package/dtslint as deprecated?

https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions <-- this could ask the user to install the proper version and thereby avoid my mistake.

**Comment by sandersn** [comment]
The README is talking about running the code in your local project where you might be developing the types, not in the DT test harness. Looking at it again, it is certainly not clear about that.

With no compile errors, you can merge, yes.

I brought up dtslint deprecation with the team and we'll discuss it. The biggest obstacle is that some people use dtslint, and `@definitelytyped/dtslint` is not a drop-in replacement&mdash;it's even more entwined with DT infrastructure, such that it's unusable outside DT.

**Comment by mrtnbrst** [comment]
Ready to merge

**Pull request merged**